### PR TITLE
Fix @interp

### PR DESCRIPTION
**Pull request by ianatol**
Before:
```
julia> using Cthulhu

julia> foo(x) = x + 2
foo (generic function with 1 method)

julia> Cthulhu.@interp foo(2)
ERROR: MethodError: no method matching mkinterp(::typeof(foo), ::Type{Tuple{Int64}})
Closest candidates are:
  mkinterp(::Core.Compiler.AbstractInterpreter, ::Any...) at ~/.julia/packages/Cthulhu/vWhSH/src/Cthulhu.jl:488
Stacktrace:
 [1] top-level scope
   @ REPL[8]:1
```

After:
```
julia> using Cthulhu

julia> foo(x) = x + 2
foo (generic function with 1 method)

julia> (interp, mi) = Cthulhu.@interp foo(2);

julia> interp.opt[mi].inferred.ir
1 1 ─ %1 = Base.add_int(_2, 2)::Int64                                                                                                                                                                            │╻ +
  └──      return %1
```

### CODE REVIEW

**Comment by simeonschaub** [comment]
Could you add a test? Otherwise LGTM

**Comment by ianatol** [comment]
Added some simple tests!

File: src/Cthulhu.jl [hunk 1]
```diff
@@ -485,6 +485,8 @@ function mkinterp(interp::AbstractInterpreter, @nospecialize(args...))
     (interp′, mi)
 end
 
+mkinterp(@nospecialize(args...)) = mkinterp(NativeInterpreter(), args...)
```
Comment (aviatesk) [inline src/Cthulhu.jl:488]:
```suggestion
mkinterp(@nospecialize(args...); interp::AbstractInterpreter=NativeInterpreter()) = mkinterp(interp, args...)
```

in order to allow something like:
```julia
julia> world = Base.get_world_counter();

julia> fsin(a) = sin(a); callf(f, args...) = f(args...);

julia> using Cthulhu

julia> Cthulhu.@interp interp=CC.NativeInterpreter(world) callf(fsin, 42)
(Cthulhu.CthulhuInterpreter(Core.Compiler.NativeInterpreter(...), Dict{Union{Core.Compiler.InferenceResult, Core.MethodInstance}, Cthulhu.InferredSource}(MethodInstance for callf(::typeof(fsin), ::Int64) => Cthulhu.InferredSource(CodeInfo(
    @ REPL[2]:1 within `callf`
1 ─     Core._apply_iterate(Base.iterate, f, args)::Union{}
), Any[Core.Compiler.UnionSplitApplyCallInfo(Core.Compiler.ApplyCallInfo[Core.Compiler.ApplyCallInfo(Core.Compiler.MethodMatchInfo(Core.Compiler.MethodLookupResult(Any[], Core.Compiler.WorldRange(0x0000000000000000, 0x0000000000007c67), false)), Union{Nothing, Core.Compiler.AbstractIterationInfo}[nothing])])], Union{})), Dict{Core.MethodInstance, Core.CodeInstance}(MethodInstance for callf(::typeof(fsin), ::Int64) => Core.1 1 ─ %1 = Core.getfield(_3, 1)::Int64::typeof(fsin), ::Int64), #undef, 0x0000000000000001, 0x0000000000007c67, Union{}, #undef, Cthulhu.OptimizedSource(                                                        │
  │        (_2)(%1)::Union{}                                                                                                                                                                                     │
  └──      unreachable                                                                                                                                                                                           │
  , CodeInfo(                                                                                                                                                                                                     
    @ REPL[2]:1 within `callf`
1 ─     Core._apply_iterate(Base.iterate, f, args)::Union{}
), false), false, false, Ptr{Nothing} @0x0000000000000000, Ptr{Nothing} @0x0000000000000000)), Dict{Core.MethodInstance, Vector{Pair{Int64, String}}}()), MethodInstance for callf(::typeof(fsin), ::Int64))
```

Comment (ianatol) [reply src/Cthulhu.jl:488]: Good point

**Pull request merged**